app/controllers/api/v1/users_controller.rb

```diff
@@ -28,7 +29,20 @@ def index
 
       api :POST, '/v1/users', "Create user"
       def create
-        save_and_render_object_with_errors(@user)
+        @user.assign_attributes(user_params)
+
+        if @user.save
+          if params["user"]["organisations_users_ids"].present?
+            @user.organisations << Organisation.find_by(id: params["user"]["organisations_users_ids"])
+          end
+
+          if params["user"]["user_role_ids"] && User.current_user.can_manage_users?
+            @user.create_or_remove_user_roles(params["user"]["user_role_ids"])
```

Comment:
I think it's time to wrap some further constraints around the `create_or_remove_user_roles` method. E.g. we need to prevent a user from being able to give roles that are more elevated than they themselves have. Let's create a separate user concern to manage this... 

```
app/models/concerns/manage_user_roles.rb

class ManageUserRoles

  def update_roles(user, role_ids)
    max_level = User.current_user.roles.max(:level)
    allowed_roles = Role.where(id: role_ids).where("level =< ?", max_level)
    ... rest of code
  end

end

```

So the idea here is that we add a new column "level" to roles.

|Role name | Level |
| Donor | 1 |
| Charity | 1 |
| Reviewer | 5 |
| Supervisor | 10 |
| Order Fulfilment | 5 |
| Order Administrator | 10 |
| Stock Fulfillment | 5 |
| Stock Administrator | 10 |
| System | 15 |
| api-write | 15 |

If you are a supervisor (level 10) then you can add the following roles to users: Donor, Charity, Reviewer, Order Fulfilment, Order Administrator, Stock Fulfilment, Stock Administrator. But NOT System or api-write